### ISSUE TITLE
[BUG] PyTorch dataloader requires: All columns required to have same data type

### ISSUE THREAD

**Issue by marcromeyn** [NVIDIA/NVTabular#917]
**Describe the bug**
`TorchAsyncItr` throws an error when not all columns have the same data-type. This for instance shows up when you have multiple labels that have different data-types. This works for TensorFlow but currently not with PyTorch.
 

**Additional context**
![image](https://user-images.githubusercontent.com/3015785/123943489-7ec69080-d99c-11eb-9aec-4defa5d556f0.png)
